docs/Timelines.md

```diff
@@ -0,0 +1,149 @@
+Timelines
+---------
+
+This document covers Timeline internals. It's meant to explain how
+timelines work so to help you understand the code and what each part
+of it contributes to the overall picture.
+
+# Fake timeline: your first timeline
+
+Previous to timelines, Nakadi would connect to a single Kafka cluster,
+which used to be specified in the application yaml file. So this was a
+static configuration loaded during boot time. Once timelines were
+introduced, in order to easy the migration process and remove special
+cases from the implementation, a fake timeline was introduced.
+
+Fake timelines exists only in memory. It's a Timeline which is not
+stored in the database. Fake timelines can be seen as a fallback
+strategy for when there is no actual timeline for a given event type.
+
+Also, fake timelines use the default storage:the on defined in the
+application configuration.
```

Comment:
It may be good to clarify here that there is at most one fake Timeline per ET, and that it is always the first one